pages/exam/index.tsx

```diff
@@ -67,16 +70,41 @@ const Exam: NextPage = () => {
   const numberOfQuestions = data?.randomQuestions.length || 0;
 
   const handleNextQuestion = (questionNo: number) => {
-    if (questionNo > numberOfQuestions) {
-      checkPassed();
-      return;
+    if (questionNo <= numberOfQuestions) {
+      if (!allQuestionsTouched) {
+        setCurrentQuestionIndex(questionNo);
+      }
+      else if (!skippedQuestions.isEmpty()) {
+        setCurrentQuestionIndex(skippedQuestions.dequeue() ?? numberOfQuestions);
+      }
+      else {
+        checkPassed();
+      }
     }
+    else {
+      setAllQuestionsTouched(true);
 
-    if (questionNo <= numberOfQuestions) {
-      setCurrentQuestionIndex(questionNo);
+      if (!skippedQuestions.isEmpty()) {
+        setCurrentQuestionIndex(skippedQuestions.dequeue() ?? numberOfQuestions);
+      }
+      else {
+        checkPassed();
+      }
     }
   };
 
+  const handleSkipQuestion = (questionNo: number) => {
+    skippedQuestions.enqueue(questionNo);
+
+    // Unset any selected answer
+    setAnswers(prevState => {
+      if (prevState[questionNo]) delete prevState[questionNo];
+      return prevState;
+    });
```

Comment:
You shouldn't modify the original object from state, instead, create a copy of this object, delete key, and finally update the state with a new object.

Reply:
Fixed in the latest commit. Thanks for pointing out